client.go

```diff
@@ -981,6 +1023,17 @@ func TLSConfig(config *tls.Config) ClientOption {
 	}
 }
 
+func SkipVersionCheck(skip bool) ClientOption {
```

Comment:
We should either call this `CheckVersion(check bool)` or `SkipVersionCheck()` (no argument). I think I prefer the latter as it makes it clear that the default behavior is to check the version, but if you pass `SkipVersionCheck()`, the version check will be skipped.

Reply:
The reason I've called it `SkipVersionCheck` instead of `CheckVersion` is, it sets `SkipVersionCheck` client option directly (which can currently be set by the user too) which in turn is directly assigned to `versionChecked` field in `Client{}` without any logic operators/conditions. `SkipVersionCheck()` without a param looks a bit strange to me. Ideally we could use `SkipVersionCheck` itself, but  that would complicate other parts of the code.

Reply:
I think if we're going to buy into the whole "functional options" thing then `SkipVersionCheck()` with no arguments is perfectly idiomatic. Ideally, these option funcs would be the only way of setting options and everything else would be unexported. I know that right now we're in the situation of supporting both ways of setting options, but I think we need to deprecate the old way.

Reply:
I agree about deprecating the old way (although it has some benefits over functional options when the options are programmatically generated). I'll change it to `SkipVersionCheck()`, but one potential problem when that flag should be set conditionally depending on the value a variable. Currently the user can just do: `client, _ := pilosa.NewClient(addr, SkipVersionCheck(someVar))`.